File: tests/java/src/main/java/com/escape/tests/cases/Case001CacheProfile.java
package com.escape.tests.cases;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Supplier;

/** Task case 001: cache profile payload for downstream business processing. */
public class Case001CacheProfile {
    private static final Map<String, Map<String, String>> RETAINED_CACHE = new HashMap<>();
    private static final List<Map<String, String>> RETAINED_AUDIT = new ArrayList<>();
    private static final List<Supplier<String>> RETAINED_HANDLERS = new ArrayList<>();

    public static String execute(String input) {
        // Task: cache profile records and prepare transport-ready payload.
        String taskName = "cache_profile";
        String raw = (input == null || input.isEmpty()) ? "sample" : input;
        Map<String, String> payload = new HashMap<>();
        payload.put("task", taskName);
        payload.put("entity", "profile");
        payload.put("stage", "cache");
        payload.put("input", raw);
        // ESCAPE: payload is promoted to class-level retained cache.
        RETAINED_CACHE.put("case_001", payload);
        return "ok";
    }
}

File: tests/go/cases/case_001_cache_profile.go
package escape_tests

var retainedCase001 = []map[string]string{}

func Case001CacheProfile(input string) string {
    // Task case 001: cache profile payload for downstream business processing.
    raw := input
    if raw == "" {
        raw = "sample"
    }
    payload := map[string]string{
        "task": "cache_profile",
        "entity": "profile",
        "stage": "cache",
        "input": raw,
    }
    // ESCAPE: payload is promoted to package-level retained cache.
    retainedCase001 = append(retainedCase001, payload)
    return "ok"
}
